olive/engine/engine.py

```diff
@@ -1014,18 +1022,19 @@ def _run_pass(
         # run pass
         host = self.host_for_pass(pass_id)
         if host.system_type != SystemType.AzureML:
-            input_model = self._prepare_non_local_model(input_model)
+            input_model_config = self._prepare_non_local_model(input_model_config)
```

Comment:
@jambayk , do we need move the model download logic to OliveSystem? if yes, how could we pass the cache directory?

Reply:
maybe we can add OliveSystem.prepare_model(cache_dir) as a signature?

Reply:
but I don't think that's high priority since we only have aml system that has a different preparation (does nothing).